src/cljc/rems/common_util.cljc

```diff
@@ -140,3 +154,37 @@
   (is (= [[:a] [:b]] (recursive-keys {:a [1] :b "foo"})))
   (is (= [[:a :b] [:a :c] [:a :d :e] [:a :d :f]]
          (recursive-keys {:a {:b 1 :c nil :d {:e "foo" :f [3]}}}))))
+
+(defn parse-int [s]
+  #?(:clj (try
+            (when s
+              (java.lang.Integer/parseInt s))
+            (catch NumberFormatException e
+              nil))
+     :cljs (let [x (js/parseInt s)]
+             (when-not (js/isNaN x)
+               x))))
+
+
+(deftest test-parse-int
+  (is (= nil (parse-int nil)))
+  (is (= nil (parse-int "")))
+  (is (= nil (parse-int "a")))
+  (is (= 7 (parse-int "7"))))
+
+(defn remove-empty-keys
+  "Given a map, recursively remove keys with empty map or nil values.
+
+  E.g., given {:a {:b {:c nil} :d {:e :f}}}, return {:a {:d {:e :f}}}."
+  [m]
+  (into {} (filter (fn [[_ v]] (not ((if (map? v) empty? nil?) v)))
+                   (mapv (fn [[k v]] [k (if (map? v)
+                                          (remove-empty-keys v)
+                                          v)])
+                         m))))
```

Comment:
could use medley's `filter-vals` and `map-kv-vals` to make this a bit more readable. not insisting.

Reply:
I inherited this part so I intended not to touch it yet. I'll add a cleanup issue about it.